File: frontend/src/Services/ConsultationService.js

import { startConsultationSession, sendConsultationMessage, getConsultationMessages, closeConsultationSession, deleteConsultationSession, getMyConsultationSessions } from './api';


const ConsultationService = {
    startSession: (doctorId, initialMessage) => startConsultationSession(doctorId, initialMessage),
    getMySessions: () => getMyConsultationSessions(),
    sendMessage: (sessionId, content) => sendConsultationMessage(sessionId, content),
    getMessages: (sessionId) => getConsultationMessages(sessionId),
    closeSession: (sessionId) => closeConsultationSession(sessionId),
    deleteSession: (sessionId) => deleteConsultationSession(sessionId)
};

export default ConsultationService; 

File: backend/hiv-treatment-medical-service-system/src/main/java/com/group7/hivcare/hivtreatmentmedicalservicesystem/consultation/service/ConsultationService.java
package com.group7.hivcare.hivtreatmentmedicalservicesystem.consultation.service;

import com.group7.hivcare.hivtreatmentmedicalservicesystem.consultation.dto.*;
import org.springframework.security.core.Authentication;
import java.util.List;

public interface ConsultationService {
    ConsultationSessionDTO startSession(StartConsultationDTO dto, Authentication authentication);

    List<ConsultationSessionDTO> getMySessions(Authentication authentication);

    ConsultationMessageDTO sendMessage(Integer sessionId, SendMessageDTO dto, Authentication authentication);

    List<ConsultationMessageDTO> getMessages(Integer sessionId);

    void closeSession(Integer sessionId, Authentication authentication);

    void deleteSession(Integer sessionId, org.springframework.security.core.Authentication authentication);
}